stagehand/src/mcp-sampling-client.ts

```diff
@@ -0,0 +1,275 @@
+import { Server } from "@modelcontextprotocol/sdk/server/index.js";
+import {
+  CreateMessageRequestSchema,
+  CreateMessageResultSchema,
+} from "@modelcontextprotocol/sdk/types.js";
+import { 
+  LLMClient, 
+  CreateChatCompletionOptions, 
+  LLMResponse, 
+  ChatMessage,
+  AvailableModel
+} from "@browserbasehq/stagehand";
+
+/**
+ * LLM Client that uses MCP Sampling to request completions from the MCP client
+ * instead of using a direct API key to an LLM provider.
+ */
+export class MCPSamplingLLMClient extends LLMClient {
+  private server: Server;
+
+  constructor(server: Server, modelName: AvailableModel = "gpt-4o", userProvidedInstructions?: string) {
+    super(modelName, userProvidedInstructions);
+    this.server = server;
+    this.type = "mcp-sampling";
+    this.hasVision = true; // Assume sampling client supports vision
+  }
+
+  async createChatCompletion<T = LLMResponse & { usage?: LLMResponse["usage"] }>(
+    options: CreateChatCompletionOptions
+  ): Promise<T> {
+    const { messages, temperature, maxTokens, requestId } = options.options;
+    const { logger } = options;
+
+    try {
+      logger({
+        category: "llm",
+        level: 1, // info
+        message: `Requesting completion via MCP Sampling for request ${requestId}`,
+        auxiliary: {
+          messages: {
+            value: messages.length.toString(),
+            type: "integer"
+          },
+          temperature: {
+            value: temperature?.toString() || "undefined",
+            type: "float"
+          },
+          maxTokens: {
+            value: maxTokens?.toString() || "undefined", 
+            type: "integer"
+          },
+        },
+      });
+
+      // Convert Stagehand chat messages to MCP sampling format
+      const mcpMessages = this.convertToMCPMessages(messages);
+
+      // Create the sampling request
+      const samplingRequest = {
+        method: "sampling/createMessage" as const,
+        params: {
+          messages: mcpMessages,
+          systemPrompt: this.getSystemPrompt(messages),
+          modelPreferences: {
+            hints: [{
+              name: this.modelName,
+            }],
+            costPriority: 0.5,
+            speedPriority: 0.5,
+            intelligencePriority: 0.8,
```

Comment:
can you explain the justification for these priority values given the use cases in stagehand?